services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreDeletionValidationUtils.java

```diff
@@ -0,0 +1,263 @@
+          return true;
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Failed to check PubSub topics for store: {} in cluster: {}", storeName, clusterName, e);
+      result.setStoreNotDeleted("Failed to check PubSub topics: " + e.getMessage());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if Helix resources for both main store and system stores still exist.
+   * This method directly queries Helix for all resources and filters for store-related ones.
+   * It does not depend on existing PubSub topics.
+   */
+  private static boolean checkAllHelixResources(
+      VeniceHelixAdmin admin,
+      StoreDeletedValidation result,
+      String clusterName,
+      String storeName) {
+    try {
+      List<String> allResources = admin.getAllLiveHelixResources(clusterName);
+
+      for (String resourceName: allResources) {
+        if (isHelixResourceRelatedToStore(resourceName, storeName)) {
+          result.setStoreNotDeleted(String.format("Helix resource still exists: %s", resourceName));
+          return true;
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Failed to check Helix resources for store: {} in cluster: {}", storeName, clusterName, e);
+      result.setStoreNotDeleted("Failed to check Helix resources: " + e.getMessage());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Determines if a Helix resource name is related to the specified store.
+   * Uses existing Venice utilities for parsing store names from version topics.
+   */
+  private static boolean isHelixResourceRelatedToStore(String resourceName, String storeName) {
+    // Use existing Version utility to check if it's a version topic
+    if (Version.isVersionTopic(resourceName)) {
+      // Use existing Version utility to parse store name
+      String resourceStoreName = Version.parseStoreFromVersionTopic(resourceName);
+
+      // Check if it's a version topic for the main store
+      if (storeName.equals(resourceStoreName)) {
+        return true;
+      }
+
+      // Check if it's a version topic for any system store
+      for (VeniceSystemStoreType systemStoreType: VeniceSystemStoreType.getUserSystemStores()) {
+        String systemStoreName = systemStoreType.getSystemStoreName(storeName);
+        if (systemStoreName.equals(resourceStoreName)) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Determines if a PubSub topic is related to the specified store.
+   * This includes version topics, real-time topics, view topics, and system store topics.
+   * Uses PubSubTopic's built-in store name validation for better accuracy.
+   * 
+   * @param topic the PubSub topic to check
+   * @param storeName the name of the store 
+   * @param systemStoreTypes the list of system store types to check against
+   * @return true if the topic is related to the store, false otherwise
+   */
+  public static boolean isStoreRelatedTopic(
```

Comment:
This method is identical to `isHelixResourceRelatedToStore`. Can we collapse them in one?